app/frontend/packs/components/checkbox_filter.js

```diff
@@ -0,0 +1,164 @@
+let CheckboxFilter = function(params) {
+  this.params = params
+  this.container = $(params.container)
+  this.container.addClass('app-checkbox-filter--enhanced')
+  this.checkboxes = this.container.find("input[type='checkbox']")
+  this.checkboxesContainer = this.container.find('.app-checkbox-filter__container')
+  this.checkboxesInnerContainer = this.checkboxesContainer.children('.app-checkbox-filter__container-inner')
+  this.legend = this.container.find('legend')
+  this.legend.addClass('govuk-visually-hidden')
+  this.setupStatusBox()
+  this.setupHeading()
+  this.setupTextBox()
+  this.setupHeight()
+}
+
+CheckboxFilter.prototype.setupHeading = function() {
```

Comment:
The prototype JS seems to work well when tested in our app and for this spike, however I wonder if we can reuse some of the JS from the original filter and leverage existing logic for to autocompletes? 
We also might need to tidy up this JS so it's more inline with the JS syntax and style we have in the app in the real implementation.